stable/prometheus/templates/_helpers.tpl

```diff
@@ -9,56 +9,110 @@ Expand the name of the chart.
 {{/*
 Create a default fully qualified app name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
 */}}
 {{- define "prometheus.fullname" -}}
+{{- if .Values.fullnameOverride -}}
+{{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
 {{- $name := default .Chart.Name .Values.nameOverride -}}
+{{- if contains $name .Release.Name -}}
+{{-  printf "%s" .Release.Name  | trunc 63 | trimSuffix "-" -}}
+{{- else -}}
 {{- printf "%s-%s" .Release.Name $name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+{{- end -}}
+{{- end -}}
 
 {{/*
 Create a fully qualified alertmanager name.
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
+If release name contains chart name it will be used as a full name.
```

Comment:
Templates other than the standard fullname template should consider the suffix. Otherwise you'd get the same name for everything.

Reply:
yep, sorry I did not update that snippet.  They actually do honor the suffix.

Reply:
updated the doc string to specify the suffix also

Reply:
I think you misunderstood. If `fullnameOverride` is set, everything will get the same name and the suffix is not considered.